fusionbox/fabric/django/new.py

```diff
@@ -143,18 +144,25 @@ def generate_pyc():
         run('python -m compileall . > /dev/null')
 
 
+@contextlib.contextmanager
+def cd_git_extract(gitref, tmp_dir=tempfile.mkdtemp()):
```

Comment:
This is a python footgun. Calling this function multiple times from the same process will mean tmp_dir is the same for each call. 

https://www.thedigitalcatonline.com/blog/2015/02/11/default-arguments-in-python/#default-arguments-evaluation

Change to `tmp_dir=None` and have an `if tmp_dir is None` guard clause in the function body.

Reply:
That footgun has definitely caused me many headaches in the past, but in this case this is the correct behavior: I want each call to `cd_git_extract` to use the same `tmp_dir`. Each different `gitref` gets it's own subdirectory: https://github.com/fusionbox/fusionbox-fabric-helpers/blob/639b55718db1a1f07a4be12eb0a5945e22307057/fusionbox/fabric/django/new.py#L150